src/Hipace.cpp

```diff
@@ -450,7 +450,6 @@ Hipace::Evolve ()
                 AMREX_ALWAYS_ASSERT(m_multi_plasma.GetNPlasmas() <= 1);
                 // Before that, the 3D fields of the envelope are not initialized (not even allocated).
                 m_multi_laser.Init3DEnvelope(step, bx, Geom(0));
-                if (it == n_boxes-1) ResetLaser();
```

Comment:
I believe this is not correct, because `Init3DEnvelope` does not modify any slice: it actually laser slice `WhichLaserSlice::np1j00` as a temporary array, but it resets it to initial value at the end. In a previous implementation, slice `WhichLaserSlice::np1j00` was not reset the the end of `Init3DEnvelope`, so once the laser was initialised, when pushing the first slice, some info of the last slice (of current box) was being used. Just one thing to keep in mind as we remove this reset.